src/main/java/sirius/db/es/Elastic.java

```diff
@@ -336,15 +338,88 @@ protected String determineIndex(EntityDescriptor ed) {
+     */
+    protected String determineWriteAlias(EntityDescriptor ed) {
+        return writeIndexTable.getOrDefault(ed, ed.getRelationName() + ACTIVE_ALIAS);
+    }
+
+    /**
+     * Resolves the effective index name being used for the given entity.
+     *
+     * @param ed the descriptor of the entity to determine the effective index for
+     * @return the effective index to which {@link #determineReadAlias(EntityDescriptor) the read alias} points
+     * @throws sirius.kernel.health.HandledException in case the alias setup failed and the expected alias does
+     *                                               not exist
+     */
+    public String determineEffectiveIndex(EntityDescriptor ed) {
+        return getLowLevelClient().resolveIndexForAlias(determineReadAlias(ed))
+                                  .orElseThrow(() -> Exceptions.handle()
+                                                               .to(Elastic.LOG)
+                                                               .withSystemErrorMessage(
+                                                                       "There is no index present for the alias (%s) of entity '%s'",
+                                                                       determineReadAlias(ed),
+                                                                       ed.getType().getName())
+                                                               .handle());
+    }
+
+    /**
+     * Creates a new write index for the given entity and installs it into the {@link #writeIndexTable}.
+     * <p>
+     * This will also install the most current mappings into the newly created index.
+     *
+     * @param ed the entity descriptor of the entity to install a new write index for
+     */
+    public void createAndInstallWriteIndex(EntityDescriptor ed) {
+        String nextIndexName = indexMappings.determineNextIndexName(ed);
+        indexMappings.createMapping(ed, nextIndexName, IndexMappings.DynamicMapping.STRICT);
+        writeIndexTable.put(ed, nextIndexName);
+    }
+
+    /**
+     * Makes the current write index of the given entity also the read index by moving the {@link #ACTIVE_ALIAS}.
+     * <p>
+     * This will also remove the write-redirection by clearing the entry in the {@link #writeIndexTable} as
+     * the indices / aliases are now the same.
+     *
+     * @param ed the entity descriptor for which the write index should also become the new read index / alias
+     */
+    public void commitWriteIndex(EntityDescriptor ed) {
+        String writeIndexName = writeIndexTable.get(ed);
+        if (writeIndexName == null) {
+            throw Exceptions.createHandled()
+                            .withSystemErrorMessage("These is no write index available for %s", ed.getType().getName())
+                            .handle();
+        }
+
+        getLowLevelClient().createOrMoveAlias(determineReadAlias(ed), writeIndexName).toJSONString();
+        writeIndexTable.remove(ed);
```

Comment:
```suggestion
        rollbackWriteIndex(ed);
```

Reply:
Technically correct, but I'd say that it looks semantically confusing to invoke rollback within commit :-/

Reply:
Possible solution is to rename `rollbackWriteIndex` to `removeWriteIndex`?